hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -1357,9 +1357,16 @@ public HFileBlock readBlock(long dataBlockOffset, long onDiskBlockSize, final bo
         BlockType.BlockCategory category = hfileBlock.getBlockType().getCategory();
         final boolean cacheCompressed = cacheConf.shouldCacheCompressed(category);
         final boolean cacheOnRead = cacheConf.shouldCacheBlockOnRead(category);
+        Optional<Boolean> cacheFileBlock = Optional.of(true);
+        // Additionally perform the time-based priority checks to see
+        // whether, or not to cache the block.
+        if (cacheConf.getBlockCache().isPresent()) {
+          cacheFileBlock = cacheConf.getBlockCache().get().shouldCacheFile(getHFileInfo(), conf);
+        }
+        final boolean shouldCacheFileBlock = cacheFileBlock.get();
 
```

Comment:
Can we move this inside cacheConf.shouldCacheBlockOnRead?

Reply:
This API shouldCacheBlockOnRead, is used at multiple places and takes only one argument (category). If I move the logic to this function, I will have to additionally pass these two arguments (getHFileInfo(), conf) to all usages. Hence, I will work on creating a new API within CacheConfig which internally checks for the data hotness and returns whether or not, to cache the file block.

Reply:
I'm fine with have multiple `shouldCache` methods, or even overloaded versions of `shouldCacheBlockOnRead`, but can't we have one `shouldCache` like method in CacheConfig that combines both checks? That way HFileReaderImpl code doesn't need to reference multiple checks and add different boolean variables each time it needs to check it.